app/components/primer/alpha/action_list.ts

```diff
@@ -21,18 +21,23 @@ export class ActionListTruncationObserver {
   }
 
   update(el: HTMLElement) {
-    const label = el.querySelector('.ActionListItem-label')
-    if (!label) return
-
-    const tooltip = el.querySelector('.ActionListItem-truncationTooltip') as HTMLElement | null
-    if (!tooltip) return
-
-    const isTruncated = label.scrollWidth > label.clientWidth
-
-    if (isTruncated) {
-      tooltip.style.display = ''
-    } else {
-      tooltip.style.display = 'none'
+    const items = el.querySelectorAll('li')
+    const labels = el.querySelectorAll('.ActionListItem-label')
+    const tooltips = el.querySelectorAll('.ActionListItem-truncationTooltip')
+
+    for (let i = 0; i < items.length; i++) {
+      const label = labels[i] as HTMLElement
+      if (!label) return
+      const tooltip = tooltips[i] as HTMLElement
+      if (!tooltip) return
+
+      const isTruncated = label.scrollWidth > label.clientWidth
+
+      if (isTruncated) {
+        tooltip.style.display = ''
+      } else {
+        tooltip.style.display = 'none'
+      }
     }
```

Comment:
Will this work as expected if some of the items don't have tooltips? Maybe something like this would work in all cases:

```suggestion
    const items = el.querySelectorAll('li')
    
    for (const item of items) {
      const label = item.querySelector('.ActionListItem-label')
      if (!label) continue
      const tooltip = item.querySelector('.ActionListItem-truncationTooltip')
      if (!tooltip) continue

      const isTruncated = label.scrollWidth > label.clientWidth

      if (isTruncated) {
        tooltip.style.display = ''
      } else {
        tooltip.style.display = 'none'
      }
    }
```